### ISSUE TITLE
Tailscale does not set up IPv6 NAT when running in containers

### ISSUE THREAD

**Issue by irbekrm** [tailscale/tailscale#11344]
### What is the issue?

It appears that in some cases (tested with GKE and EKS deployments), tailscale running in non-userspace mode in containers incorrectly determines that it is not possible to set up IPv6 NAT rules ([masquerading](https://github.com/tailscale/tailscale/blob/v1.60.1/util/linuxfw/iptables_runner.go#L417-L427)). 

This seems to be because [the check for whether IPv6 NAT is available](https://github.com/tailscale/tailscale/blob/main/util/linuxfw/linuxfw.go#L173-L193) always (? as far as I have tested) fails in container envs:
- `/proc/net/ip6_tables_names` seems to remain empty till the first time `iptables` rules are queried (see https://github.com/moby/moby/issues/34337 and my testing seemed to confirm that)
- modprobe does not work in container envs (unless running as privileged)

One issue that this creates (from a user report) is that it is not possible to access IPv6 hosts via exit nodes deployed on Kubernetes in non-userspace mode.
I tested that an IPv6 masquerading rule can be added by hand to the exit node Pod and it is then able to route IPv6 traffic.

### Steps to reproduce

- Create an exit node on Kubernetes, i.e via [`Connector`](https://tailscale.com/kb/1236/kubernetes-operator?q=kubernetes%20operator#deploying-exit-nodes-and-subnet-routers-on-kubernetes-using-connector-custom-resource). 

- Exec to the Pod and observe that masquerading is only set up for IPv4 (assuming this runs with iptables):

<details>
<code>
/ # iptables -t nat -L
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination         

Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination         
ts-postrouting  all  --  anywhere             anywhere            

Chain ts-postrouting (1 references)
target     prot opt source               destination         
MASQUERADE  all  --  anywhere             anywhere             mark match 0x40000/0xff0000
/ # ip6tables -t nat -L
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination         

Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination         

</code>
</details>

- Observe that a tailnet node that uses this exit node cannot access IPv6 endpoints.

- Exec to the Pod and manually add masquerading for IPv6:

```
ip6tables -t nat -I POSTROUTING 1 -m mark --mark 0x40000/0xff0000 -j MASQUERADE
```

- Observe that IPv6 endpoints can now be accessed via the exit node.

**Comment by irbekrm** [comment]
This should have been fixed in https://github.com/tailscale/tailscale/pull/11353

`unstable-v1.61.87` image tags/`1.61.87` helm chart has the fix.

**Comment by irbekrm** [comment]
Closing as should be fixed in #11353
